packages/engine/Source/DataSources/Entity.js

```diff
@@ -494,6 +497,21 @@ Object.defineProperties(Entity.prototype, {
   wall: createPropertyTypeDescriptor("wall", WallGraphics),
 });
 
+/**
+ * Add the specified type and construct the properties for it in the Entity class
+ * @private
+ * @param {string} propertyName
+ * @param {any} Type
```

Comment:
Strictly speaking this would be
```suggestion
 * @param {object.constructor} Type
```
right?

Reply:
I know this is marked as private, but typically we do add descriptions for each parameter. That applies here and throughout this PR.

Reply:
More and more often I go down rabbit holes dug by JSDoc being old and not maintained.
`object.constructor` is not a type. After a quick google search I came across this issue https://github.com/jsdoc/jsdoc/issues/1349 which, paired with other JSDoc knowledge, seems to indicate there's no way to really specify "any _class_" as a type. The best I think we can do is specify a type of `{{ constructor: function }}` to try and only accept objects with a constructor function?